### DISCUSSION TITLE
issue: FieldArray append with React Select not working correctly.

### DISCUSSION THREAD

**Discussion by tomas1646** [react-hook-form/react-hook-form#12016] [Q&A]
### Version Number

7.51.5

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/great-shaw-pwcykw?file=%2Fpackage.json%3A10%2C31

### Steps to reproduce

1. Open Console
2. Change ReactSelectInput to "Chocolate". `dirtyFields` will be: 
```
{ ReactSelect: true }
```
3. Click on "Add Field Array Item +".
4. See error on `dirtyFields`:
```
{
    "ReactSelect": {
        "value": true,
        "label": true
    },
    "fieldArray": [
        {
            "name": true
        }
    ]
}
```

### Expected behaviour

The expected `dirtyFields` output in step 4 should be:
```
{
    "ReactSelect": true,
    "fieldArray": [
        {
            "name": true
        }
    ]
}
```

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by JokinAbarzua** [comment]
Same issue here
